Fix ScriptEditor inline colors float handling

kitbdev opened the pull request:
- Fixes https://github.com/godotengine/godot/issues/107725

Mostly whitespace changes since I tried to make the function easier to read.

`split_floats` just gives `0.0` when it encounters non-numbers, so I removed it and now check `is_valid_float` for each parameter, with a maximum of 4.
It must be a float literal, variables and expressions are not allowed here. It may be possible to use Expression to handle them, but it would still need to be readonly.

Also added checks when typing a string parameter to only show the color if it is valid and to prevent some invalid color errors (not all, see https://github.com/godotengine/godot/issues/89357).

This function will need to be moved to be language independent in the future, then it can be parsed better with GDScript.

kitbdev committed:
Fix ScriptEditor inline colors float handling

KoBeWi commented on editor/plugins/script_text_editor.cpp:
```diff
@@ -411,76 +411,91 @@ Array ScriptTextEditor::_inline_object_parse(const String &p_text, int p_line) {
+			if (stripped.length() > 0 && (stripped[0] == '\"')) {
+				// String constructor.
+				const String color_string = stripped.substr(1, stripped.length() - 2);
+				if (Color::html_is_valid(color_string) || Color::find_named_color(color_string) != -1) {
```
Checking for valid color causes the marker to get toggled repeatedly.

https://github.com/user-attachments/assets/e0818780-5f0b-482a-8fda-a5e66d879140

I think if it's inside string, it doesn't matter if the color code is valid, because it can't mean anything else, right? 🤔 (as opposed to the original issue, where a variable is treated as a number)

kitbdev commented:
Updated to show default color if the string isn't valid so it doesn't keep dissapearing and reappearing while typing.
Added more checks to prevent these cases being considered valid:
```gdscript
Color(")
Color("aaa)
Color(0xawaaaaaa)
Color(0x0 + aaaa)
Color("ff"+red)
```

KoBeWi reviewed (approved):
Works fine now.
Needs rebase.
